### DISCUSSION TITLE
How to implement my own UI for firebaseAuth?

### DISCUSSION THREAD

**Discussion by Darren120** [gladly-team/next-firebase-auth#306] [Q&A] [answered]
There is too much abstraction I am hella confused. So we are using the FirebaseAuth in the example, how do I strip that down so I can call the next-firebase-auth API when the user clicks the submit form using my custom UI and not FirebaseAuth?

OK I implemented custom UI, and it worked! ? looks like this library did some magic behind the scenes, idk what triggers the cookie collection when logged in since I didn't trigger anything from next-firebase-auth in my ui. would like some explanation its really cool

**Comment by Darren120** [comment, marked as answer]
solved

**Reply by Josem1801** [reply]
I have the same question, can you show me how?
